__tests__/auth-routes.test.js

```diff
@@ -0,0 +1,112 @@
+
+    describe(`${userType} users`, () => {
+
+      it('can create one', async () => {
+
+        const response = await mockRequest.post('/signup').send(users[userType]);
+        const userObject = response.body;
+
+        expect(response.status).toBe(201);
+        expect(userObject.token).toBeDefined();
+        expect(userObject.user._id).toBeDefined();
+        expect(userObject.user.username).toEqual(users[userType].username)
+
+      });
+
+      it('can signin with basic', async () => {
+
+        const response = await mockRequest.post('/signin')
+          .auth(users[userType].username, users[userType].password);
+
+        const userObject = response.body;
+        expect(response.status).toBe(200);
+        expect(userObject.token).toBeDefined();
+        expect(userObject.user._id).toBeDefined();
+        expect(userObject.user.username).toEqual(users[userType].username)
+
+      });
+
+      it('can signin with bearer', async () => {
+
+        // First, use basic to login to get a token
+        const response = await mockRequest.post('/signin')
+          .auth(users[userType].username, users[userType].password);
+
+        const token = response.body.token;
+
+        // First, use basic to login to get a token
+        const bearerResponse = await mockRequest
+          .get('/users')
+          .set('Authorization', `Bearer ${token}`)
+
+        // Not checking the value of the response, only that we "got in"
+        expect(bearerResponse.status).toBe(200);
+
+      });
+
+    });
+
+    describe('bad logins', () => {
+      it('basic fails with known user and wrong password ', async () => {
+
+        const response = await mockRequest.post('/signin')
+          .auth('admin', 'xyz')
+        const userObject = response.body;
+
+        expect(response.status).toBe(403);
+        expect(userObject.user).not.toBeDefined();
+        expect(userObject.token).not.toBeDefined();
+
+      });
+
+      it('basic fails with unknown user', async () => {
+
+        const response = await mockRequest.post('/signin')
+          .auth('nobody', 'xyz')
+        const userObject = response.body;
+
+        expect(response.status).toBe(403);
+        expect(userObject.user).not.toBeDefined();
+        expect(userObject.token).not.toBeDefined()
+
+      });
+
+      it('bearer fails with an invalid token', async () => {
+
+        // First, use basic to login to get a token
+        const bearerResponse = await mockRequest
+          .get('/users')
+          .set('Authorization', `Bearer foobar`)
+
+        // Not checking the value of the response, only that we "got in"
+        expect(bearerResponse.status).toBe(403);
+
+      })
+    })
+
+  });
+
+});
+
+
+// AUTH Routes
+// POST /signup creates a new user and sends an object with the user and the token to the client
+// POST /signin with basic authentication headers logs in a user and sends an object with the user and the token to the client
```

Comment:
These tests look really great, I would just like to see tests for the v1 or v2 routes